pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -113,6 +113,19 @@ func convertService(svc v1.Service, domainSuffix string) *model.Service {
 	}
 	sort.Strings(serviceaccounts)
 
+	var lbAddresses []string
+	if svc.Spec.Type == v1.ServiceTypeLoadBalancer && len(svc.Status.LoadBalancer.Ingress) > 0 {
+		lbAddresses = make([]string, len(svc.Status.LoadBalancer.Ingress))
+		for i, addr := range svc.Status.LoadBalancer.Ingress {
+			if len(addr.IP) > 0 {
+				lbAddresses[i] = addr.IP
+			}
+			if len(addr.Hostname) > 0 {
```

Comment:
else if

Reply:
It's actually an overwrite logic preferring hostname if specified